src/ExpressionEvaluator/CSharp/Test/ResultProvider/ValueFormattingTests.cs

```diff
@@ -214,8 +214,8 @@ public void String()
             Assert.Equal("\"\\uffff\"", FormatValue(s, useHexadecimal: true));
 
             string multiByte = "\ud83c\udfc8";
-            Assert.Equal(string.Format(format, "🏈"), FormatValue(multiByte));
-            Assert.Equal(string.Format(format, "🏈"), FormatValue(multiByte, useHexadecimal: true));
+            Assert.Equal(string.Format(format, "\\ud83c\\udfc8"), FormatValue(multiByte));
+            Assert.Equal(string.Format(format, "\\ud83c\\udfc8"), FormatValue(multiByte, useHexadecimal: true));
```

Comment:
@ufcpp While it is true that emoji are printable, they cannot be represented as characters without being encoded, either as a surrogate pair or as a sequence of ASCII characters representing a C# unicode escape sequence. This PR proposes them to be escaped using a C# unicode escape sequence.

Leaving them as a surrogate pair is problematic, as the resulting source cannot be encoded as UTF8. That's because surrogate characters cannot properly be encoded in UTF-8. See http://unicodebook.readthedocs.io/issues.html#strict-utf8-decoder

To enable the resulting source to be output to a file in any encoding (including UTF-8), we prefer the C# escape sequence.